src/main/java/com/programmers/springbootbasic/infrastructure/IO/Console.java

```diff
@@ -0,0 +1,74 @@
+package com.programmers.springbootbasic.infrastructure.IO;
+
+import com.programmers.springbootbasic.presentation.IOManager;
+import com.programmers.springbootbasic.presentation.validation.InputValidator;
+import jakarta.annotation.PostConstruct;
+import java.util.Scanner;
+import org.springframework.stereotype.Component;
+
+@Component
+public class Console implements IOManager {
+    private final Scanner scanner = new Scanner(System.in);
+    private InputValidator<String> stringValidator;
+    private InputValidator<String> integerValidator;
+
+    @PostConstruct
+    private void setValidator() {
+        InputValidator<String> stringValidator = new InputValidator<>();
+        stringValidator.addValidator(input -> {
+            if(input == null || input.isEmpty()) {
+                return "입력 값이 없습니다.";
+            }
+            return null;
+        });
+        this.stringValidator = stringValidator;
+
+        InputValidator<String> integerValidator = new InputValidator<>();
+        integerValidator.addValidator(input -> {
+            if(input == null || input.isEmpty()) {
+                return "입력 값이 없습니다.";
+            }
+            return null;
+        });
+        integerValidator.addValidator(input -> {
+            try {
+                Integer.parseInt(input);
+                return null;
+            } catch (NumberFormatException e) {
+                return "숫자가 아닙니다.";
+            }
+        });
+        this.integerValidator = integerValidator;
+    }
```

Comment:
```suggestion
    private void setValidator() {
        InputValidator<String> stringValidator = new InputValidator<>();
        stringValidator.addValidator(Console::validateString);
        this.stringValidator = stringValidator;

        InputValidator<String> integerValidator = new InputValidator<>();
        integerValidator.addValidator(Console::validateString);
        integerValidator.addValidator(Console::validateInteger);
        this.integerValidator = integerValidator;
    }

    private static String validateString(String input) {
        if(input == null || input.isEmpty()) {
            return "입력 값이 없습니다.";
        }
        return null;
    }

    private static String validateInteger(String input) {
        try {
            Integer.parseInt(input);
            return null;
        } catch (NumberFormatException e) {
            return "숫자가 아닙니다.";
        }
    }
```

공통 로직은 메서드 분리로 중복제거하는 것이 필요해보입니다.

Reply:
[60055d0](https://github.com/prgrms-be-devcourse/springboot-basic/pull/902/commits/60055d0c6e0c81366ef7abf5fd1449a99e65e156)

콘솔에서 사용하는 유효성 검사기들을 빈으로 등록하여 사용하도록 바꾸어 보았습니다!
다른 라이브러리에서 사용할때마다 Validator 를 만들고, Validate 하는 함수를 add 해서 등록 방식으로 사용하는 걸 보고 좋을 것 같다는 생각에 가져와본 것인데! 생각보다 중복 로직이 많이 생길 수 있어 관리가 필요한 방식이란 생각이 듭니다...